frontend/src/components/FilterPills.js

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faTimesCircle } from '@fortawesome/free-solid-svg-icons';
+import moment from 'moment';
+import { formatDateRange } from './DateRangeSelect';
+
+const filterProp = PropTypes.shape({
+  topic: PropTypes.string,
+  condition: PropTypes.string,
+  query: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
+  id: PropTypes.string,
+});
+
+/* Pill */
+export function Pill({ filter, isLast, onRemoveFilter }) {
+  const {
+    id,
+    topic,
+    condition,
+    query,
+  } = filter;
+
+  const filterNameLookup = [
+    {
+      topic: 'role',
+      display: 'Specialist',
+    },
+    {
+      topic: 'startDate',
+      display: 'Date Range',
+    },
+  ];
+
+  const determineFilterName = () => {
+    const topicMatch = filterNameLookup.find((f) => f.topic === topic);
+
+    if (topicMatch) {
+      return topicMatch.display;
+    }
+
+    return topic;
+  };
+
+  const determineQuery = (keepOriginalLength = true) => {
+    if (topic === 'startDate') {
+      if (query.includes('-')) {
+        return formatDateRange({
+          string: query,
+          withSpaces: false,
+        });
+      }
+      return moment(query, 'YYYY/MM/DD').format('MM/DD/YYYY');
+    } if (topic === 'role') {
```

Comment:
should this be an else? otherwise, might need be bumped down on its own line

Reply:
No need for an else here as all paths above return. I have put the if statement on its own line. I have also lower cased the condition.